orion/controllers/programs/programs_test.go

```diff
@@ -11,9 +13,55 @@ func TestExample(t *testing.T) {
   }
 }
 
-func TestCheckProgram(t *testing.T) {
-  isValid := CheckValidProgram()
-  if (isValid == false) {
-    t.Errorf("Check was incorrect, got: %t, expected: %t.", isValid, false)
-  }
+func Rand(a, b int) uint {
+	return uint(rand.Int() % (b - a) + a)
 }
+
+func TestCheckProgram(t *testing.T) {
+	// success examples
+	var s string
+	for i := 0; i < 1000; i++ {
+		s += "Word "
+		var grade1, grade2 uint = Rand(1, 12), Rand(1, 12)
+		if grade1 > grade2 {
+		    grade1, grade2 = grade2, grade1
+		}
+		program := Program{ProgramId: string(i), Name: s, Grade1: grade1, Grade2: grade2, Description: "a"}
+		if err := CheckValidProgram(program); err != nil {
+			fmt.Println(err)
+			return
+		}
+	}
+	
+	// failure examples
+	program := Program{ProgramId: "", Name: "Asdf", Grade1: 1, Grade2: 12, Description: "a"}
+	if err := CheckValidProgram(program); err == nil {
+		fmt.Println("Fail")
```

Comment:
When things fail, you should be calling t.Errorf(...). This will alert the testing framework that something failed.